frontend/src/hooks/useSearch.ts

```
@@ -0,0 +1,115 @@
+const buildILikeParams = <T>({ params, searchText }: TBuildParamProps<T>) =>
+  params?.reduce((acc, field) => {
+    acc[field] = { contains: searchText };
+
+    return acc;
+  }, {} as Record<TFilterStringFields<T>, unknown>);
+const buildEqInitialParams = <T>({
+  initialParams,
+}: TBuildInitialParamProps<T>) =>
+  initialParams?.reduce(
+    (acc, obj) => ({
+      ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)